accounts-db/src/accounts_db.rs

```diff
@@ -6809,27 +6809,22 @@ impl AccountsDb {
         &self,
         slot: Slot,
         accounts_and_meta_to_store: &impl StorableAccounts<'b>,
-        txn_iter: Box<dyn std::iter::Iterator<Item = &Option<&SanitizedTransaction>> + 'a>,
+        mut txn_iter: impl Iterator<Item = &'a SanitizedTransaction>,
     ) -> Vec<AccountInfo> {
-        let mut write_version_producer: Box<dyn Iterator<Item = u64>> =
-            if self.accounts_update_notifier.is_some() {
-                let mut current_version = self
-                    .write_version
-                    .fetch_add(accounts_and_meta_to_store.len() as u64, Ordering::AcqRel);
-                Box::new(std::iter::from_fn(move || {
-                    let ret = current_version;
-                    current_version += 1;
-                    Some(ret)
-                }))
-            } else {
-                Box::new(std::iter::empty())
-            };
+        let mut write_version_producer = if self.accounts_update_notifier.is_some() {
```

Comment:
we either iterate over a range or have an empty. 0..0 should be equivalent, but match the Range type so we can avoid doing a Box allocation